src/app/vendors/selectors.ts

```diff
@@ -1,3 +1,52 @@
+import { currentAccountSelector } from 'app/accounts/selectors';
+import { mergeCollectibles } from 'app/inventory/d2-stores';
+import {
+  bucketsSelector,
+  profileResponseSelector,
+  sortedStoresSelector,
+} from 'app/inventory/selectors';
+import { getCurrentStore } from 'app/inventory/stores-helpers';
+import { d2ManifestSelector } from 'app/manifest/selectors';
 import { RootState } from 'app/store/types';
+import { emptyArray, emptyObject } from 'app/utils/empty';
+import { DestinyCollectibleComponent } from 'bungie-api-ts/destiny2';
+import { createSelector } from 'reselect';
+import { D2VendorGroup, toVendorGroups } from './d2-vendors';
 
 export const vendorsByCharacterSelector = (state: RootState) => state.vendors.vendorsByCharacter;
+
+export const mergedCollectiblesSelector = createSelector(
+  profileResponseSelector,
+  (profileResponse) =>
+    profileResponse
+      ? mergeCollectibles(
+          profileResponse.profileCollectibles,
+          profileResponse.characterCollectibles
+        )
+      : emptyObject<{
+          [x: number]: DestinyCollectibleComponent;
+        }>()
```

Comment:
I suppose this could be typed as `ReturnType<typeof mergeCollectibles>`?

Reply:
fair, i avoid returntype when it can make things too flexible but in this case it's an appropriate match in the ternary